catalog/confluence/catalog-confluence-source/src/main/java/org/codice/ddf/confluence/source/ConfluenceInputTransformer.java

```diff
@@ -0,0 +1,333 @@
+        if (body != null) {
+            String cleanedText = body.toString()
+                    .replaceAll("<.*?>", " ");
+            String description = cleanedText;
+            if (description.length() > DESCRIPTION_SIZE) {
+                description = description.substring(0, DESCRIPTION_SIZE) + "...";
+            }
+            if (StringUtils.isNotEmpty(description)) {
+                String xmlSafeString = StringEscapeUtils.escapeXml10(cleanedText);
+                metacard.setAttribute(Metacard.DESCRIPTION, description);
+                metacard.setAttribute(Metacard.METADATA,
+                        String.format(PRODUCT_XML, confluenceType, xmlSafeString));
+                metacard.setAttribute(Confluence.BODY_TEXT, xmlSafeString);
+            }
+        }
+        if (confluenceType.equals("attachment")) {
+            Object comment = getJsonElement(json, "metadata", "comment");
+            if (comment != null) {
+                metacard.setAttribute(Metacard.DESCRIPTION, comment.toString());
+            }
+            Object mediaType = getJsonElement(json, "metadata", "mediaType");
+            if (mediaType != null) {
+                metacard.setAttribute(Media.TYPE, mediaType.toString());
+            }
+        } else {
+            metacard.setAttribute(Media.TYPE, "text/html");
+        }
+    }
+
+    private void parseRestrictions(MetacardImpl metacard, Object json)
+            throws CatalogTransformerException {
+        Object restrictions = getJsonElement(json, "restrictions", "read", "restrictions");
+        if (restrictions != null) {
+            ArrayList<String> userRestrictions = new ArrayList<>();
+            ArrayList<String> groupRestrictions = new ArrayList<>();
+
+            getJsonArray(restrictions, "user", "results").stream()
+                    .forEach(e -> userRestrictions.add(getStringOrDefault(e, UNKNOWN, "username")));
+            getJsonArray(restrictions, "group", "results").stream()
+                    .forEach(e -> groupRestrictions.add(getStringOrDefault(e, UNKNOWN, "name")));
+            metacard.setAttribute(Security.ACCESS_INDIVIDUALS, userRestrictions);
+            metacard.setAttribute(Security.ACCESS_GROUPS, groupRestrictions);
+        }
+    }
+
+    private Map<String, Object> getJsonObject(InputStream stream)
+            throws CatalogTransformerException {
+        String jsonString = null;
+        try {
+            jsonString = IOUtils.toString(stream);
+            Map<String, Object> rootObject = MAPPER.parser()
+                    .parseMap(jsonString);
+
+            LOGGER.debug(jsonString);
+            return rootObject;
+        } catch (IOException | RuntimeException re) {
+            throw new CatalogTransformerException("Invalid json. Could not parse: " + jsonString,
+                    re);
+        }
+    }
+
+    private List getJsonArray(Object object, String... keys) throws CatalogTransformerException {
```

Comment:
I think you might be able to remove some of this helper code if you use Boon path expressions.

https://github.com/boonproject/boon/wiki/Boon-path-expressions